ethers-providers/src/rpc/transports/ws/mod.rs

```diff
@@ -50,6 +50,34 @@ impl WsClient {
         Ok(this)
     }
 
+    /// Establishes a new websocket connection. This method allows specifying a custom websocket
+    /// configuration.
+    pub async fn connect_with_config(
+        conn: impl Into<ConnectionDetails>,
+        config: impl Into<WebSocketConfig>,
+    ) -> Result<Self, WsClientError> {
+        let (man, this) = RequestManager::connect_with_config(conn.into(), config.into()).await?;
+        man.spawn();
+        Ok(this)
+    }
+
+    /// Establishes a new websocket connection with auto-reconnects. This method allows specifying a
+    /// custom websocket configuration.
+    pub async fn connect_with_reconnects_and_config(
+        conn: impl Into<ConnectionDetails>,
+        reconnects: usize,
```

Comment:
nit:
make last argument so arg 1+2 are identical to `connect_with_config`

Reply:
good point, done in last commit, I've also changed the name so its `connect_with_config_and_reconnects` which will make it pop in IDE hints once you already decided you want to use a config